BrainPortal/lib/license_custom.rb

```diff
@@ -0,0 +1,157 @@
+# McGill University
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#license_for_groups.rb
+
+
+
+# Module containing common methods for set and access the user create, custom user-agreements
+# license agreements. Differently from the original, user can create his own licences
+module LicenseCustom
+
+  Revision_info=CbrainFileRevision[__FILE__] #:nodoc:
+
+  require 'securerandom'
+
+  # Check that the the class this module is being included into is a valid one.
+  def self.included(includer) #:nodoc:
+    unless includer <= ApplicationRecord
+      raise "#{includer} is not an ActiveRecord model. The LicenseCustom module cannot be used with it."
+    end
+
+    includer.class_eval do
+      # License agreement is a pseudo attributes and cannot be accessed if the object is not saved.
+      validate   :valid_license_agreements?
+      after_save :register_license_agreements
+    end
+  end
+
+  # Returns the list of license agreements that exists for this object.
+  def license_agreements
+    @license_agreements ||= load_license_agreements
+  end
+
+  # Sets the list of license agreements that exists for this object.
+  # Can be provided with an array of license names, or a single string with
+  # a space-or-comma-separated list of license names.
+  def license_agreements=(agreements)
+    license_agreements # loads them if not already loaded
+    agrs = agreements
+    unless agrs.is_a? Array
+      agrs = agrs.to_s.split(/[,\s]+/)
+    end
+    agrs = agrs.map { |a| a.sub(/\.html\z/, "").gsub(/[^\w-]+/, "") }.uniq.sort
+    @license_agreements = agrs
+  end
+
+  # create a license file
+  def create_license_file(content, user, basename=nil)
+    basename ||= "license_#{SecureRandom.uuid().to_s}.txt"
+    userfile = create_file(content, user, basename)
+    license_agreements << userfile.id.to_s
+    register_license_agreements
+    # post save trigger
+  end
+
+  protected
+
+  # Loads the licenses from the meta data store.
+  def load_license_agreements
+    @license_agreements           = self.new_record? ? [] : self.meta[:license_agreements].presence || []
+    @_license_agreements_original = @license_agreements.dup # to determine if anything changed when saving
+    @license_agreements
+  end
+
+  # Returns true if the set of licenses are identifiers that
+  # properly match files on the filesystem, in public/licenses/{name}.html
```

Comment:
I am not sure about what you try to achieve here. 
Here is my comments: 
- First of all, you never look at `public/licenses/{name}.html` in this method.
- Variable `invalid_licenses` seem to contain VALID license. 
- SInce `valid_license_agreements?` is a validate call back it prevents me to edit the group.

Reply:
@MontrealSergiy I still have this issue, it prevent me to save any edit to a project.